src/Fixer/LanguageConstruct/GetClassToClassKeywordFixer.php

```diff
@@ -0,0 +1,157 @@
+            [
+                new VersionSpecificCodeSample(
+                    "<?php\nget_class(\$a);\n",
+                    new VersionSpecification(80000)
+                ),
+                new VersionSpecificCodeSample(
+                    "<?php\n\n\$date = new \\DateTimeImmutable();\n\$class = get_class(\$date);\n",
+                    new VersionSpecification(80000)
+                ),
+            ],
+            null,
+            'Risky if the `get_class` function is overridden.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before MultilineWhitespaceBeforeSemicolonsFixer.
+     * Must run after NoSpacesAfterFunctionNameFixer, NoSpacesInsideParenthesisFixer.
+     */
+    public function getPriority(): int
+    {
+        return 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return \PHP_VERSION_ID >= 80000 && $tokens->isAllTokenKindsFound([T_STRING, T_VARIABLE]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky(): bool
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        for ($index = $tokens->count() - 1; $index > 0; --$index) {
+            if (!$tokens[$index]->equals([T_STRING, 'get_class'], false)) {
+                continue;
+            }
+
+            if (!(new FunctionsAnalyzer())->isGlobalFunctionCall($tokens, $index)) {
+                continue;
+            }
+
+            $braceOpenIndex = $tokens->getNextMeaningfulToken($index);
+            $braceCloseIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $braceOpenIndex);
+
+            if ($braceCloseIndex === $tokens->getNextMeaningfulToken($braceOpenIndex)) {
+                // get_class with no arguments
+                continue;
+            }
+
+            $meaningfulTokensCount = 0;
+            $variableTokensIndices = [];
+
+            for ($i = $braceOpenIndex + 1; $i < $braceCloseIndex; ++$i) {
+                if (!$tokens[$i]->equalsAny([[T_WHITESPACE], [T_COMMENT], [T_DOC_COMMENT], '(', ')'])) {
+                    ++$meaningfulTokensCount;
+                }
+
+                if (!$tokens[$i]->isGivenKind(T_VARIABLE)) {
+                    continue;
+                }
+
+                if ('$this' === strtolower($tokens[$i]->getContent())) {
+                    // get_class($this)
+                    continue 2;
+                }
+
+                $variableTokensIndices[] = $i;
+            }
+
+            if ($meaningfulTokensCount > 1 || 1 !== \count($variableTokensIndices)) {
+                // argument contains more logic, or more arguments, or no variable argument
+                continue;
+            }
+
+            $this->fixGetClassCall($tokens, $index, $braceOpenIndex, $braceCloseIndex, current($variableTokensIndices));
```

Comment:
here, collect cases to fix only and fix them with calling `->insertSlices` once outside of main loop (`for ($index = $tokens->count() - 1; $index > 0; --$index) {`)

Reply:
@keradus I'm not sure if I captured what you want here in my latest commit

Reply:
you are dealing with multiple insertions.
you are calling `fixGetClassCall` multiple times inside `applyFix`, and each of this time is calling `insertAt`.
instead of doing that, call `insertSlices` ONCE (outside of main loop) in `applyFix`.

If you would call `insertAt` 20 times, you will rebuild tokens internal indexes 20 times.
Calling `insertSlices` once will mean you need to rebuild tokens internal indexes only once.